### ISSUE TITLE
Data loss when multi master replica is active

### ISSUE THREAD

**Issue by nekulin** [JohnSully/KeyDB#142]
I set up an active replica between 3 nodes, I have data on 1 server every other time, although there is data on 2 other servers, why can this happen ?

**Comment by Talkabout** [comment]
I think, based on the documentation, 3 nodes require master/master replication instead of active replication. Have you tried that?

**Comment by nekulin** [comment]
What is the most reliable replication to choose ? How many nodes do I need for active replication to work, I can be sure that the data will be on all nodes ?

**Comment by nekulin** [comment]
How do I understand that I should only use active replication with 2 nodes ?

**Comment by Talkabout** [comment]
Hi @nekulin ,

yes, as far as I have understood the documentation for KeyDB, active replication is limited to 2 nodes only. This is also how I am using it. When you need to synchronize more than 2 nodes, master/master replication is the way to go. But the documentation also states, that this kind of sync (> 2 nodes) is not stable yet.

**Comment by JohnSully** [comment]
Are you using a ring or mesh topology?  

I’ve added a number of fixes for the ring topology to the unstable branch.

**Comment by nekulin** [comment]
> Are you using a ring or mesh topology?
> 
> I’ve added a number of fixes for the ring topology to the unstable branch.

By default )

**Comment by nekulin** [comment]
> Hi @nekulin ,
> 
> yes, as far as I have understood the documentation for KeyDB, active replication is limited to 2 nodes only. This is also how I am using it. When you need to synchronize more than 2 nodes, master/master replication is the way to go. But the documentation also states, that this kind of sync (> 2 nodes) is not stable yet.

Is it stable ?

**Comment by Talkabout** [comment]
I am still in testing phase, at the moment I am not facing any issues. To be noted here: I am using KeyDB as distributed cache, so minor sync issues are not that problematic for me. You should try yourself to assure it suits your needs.

**Comment by nekulin** [comment]
I caught a couple of times problems with the connection, strange

**Comment by JohnSully** [comment]
Multimaster is still experimental although I’m hoping to get it stable fairly soon. 

Originally I had though Active Rep wouldn’t be useful without multimaster but it turned out many people were happy with just 2 nodes. That led me to focus just on 2 node Active Rep for the last few months.

**Comment by JohnSully** [comment]
You should also be aware of bug #111 which is also an issue with Multi Master

**Comment by JohnSully** [comment]
@nekulin Can you provide steps to reproduce the issues you are seeing. It’s hard for me to resolve the issue if I don’t know what problems your hitting.

**Comment by JohnSully** [comment]
This should be fixed in v5.3.2 where I've done a broad refactoring and testing of Active Replication (and multimaster).  If the issue appears in 5.3.2 or above please open a new issue.

**Comment by rhzs** [comment]
Hi @JohnSully , i tried to use KeyDB on Multi master and active replica using GKE and Helm Chart v3 https://github.com/Enapter/charts/tree/master/keydb

I can't get data on different KeyDB pod.  I have opened issue Helm chart implementation as well on https://github.com/Enapter/charts/issues/2

Here is my log and screenshot using `keydb-cli`
```
❯ kubectl run -it keydb-cli --image=eqalpha/keydb --restart=Never /bin/bash
If you don't see a command prompt, try pressing enter.
root@keydb-cli:/data# keydb-cli -h 10.8.1.xxx -p 6379 monitor
OK
^C
root@keydb-cli:/data# keydb-cli -h 10.8.1.xxx -p 6379
Message of the day:
  We want to hear from you! Help make KeyDB better with a quick 5 question survey: https://www.surveymonkey.com/r/Y9XNS93

10.8.1.xxx:6379> SET abc def;
OK
10.8.1.xxx:6379> exit
root@keydb-cli:/data# keydb-cli -h 10.8.0.xxx -p 6379
Message of the day:
  We want to hear from you! Help make KeyDB better with a quick 5 question survey: https://www.surveymonkey.com/r/Y9XNS93

10.8.0.xxx:6379> GET abc
(nil)
10.8.0.xxx:6379> exit
root@keydb-cli:/data# keydb-cli -h 10.8.1.xxx -p 6379 monitor
OK



^C
root@keydb-cli:/data# keydb-cli -h 10.8.0.xxx -p 6379 monitor
OK
^C
```

I should be able to get `def` on `10.8.1.xxx`, right?
